flang/runtime/tools.cpp

```diff
@@ -173,5 +173,17 @@ RT_API_ATTRS void ShallowCopy(const Descriptor &to, const Descriptor &from) {
   ShallowCopy(to, from, to.IsContiguous(), from.IsContiguous());
 }
 
+RT_API_ATTRS const char *EnsureNullTerminated(
+    const char *str, size_t length, Terminator &terminator) {
+  if (length <= std::strlen(str)) {
```

Comment:
Don't call `std::strlen()` on a string that is not known to be NUL-terminated.

Reply:
done, used `std::memchr(str, '\0', length)` instead